web/src/features/VersionDiff/ViewDiffButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { Version } from "@types";
+import { useSelectedApp } from "@features/App/hooks/useSelectedApp";
+import { useIsHelmManaged } from "@components/hooks";
+
+interface Props {
+  onWhyNoGeneratedDiffClicked: (version: Version) => void;
+  onWhyUnableToGeneratedDiffClicked: (version: Version) => void;
+  onViewDiffClicked: (firstSequence: number, secondSequence: number) => void;
+  version: Version;
+  versionHistory: Version[];
+}
+
+interface DiffSummary {
+  filesChanged?: number;
+}
+
+// TODO: unmarshal this in the fetch handler
+function unmarshallDiffSummary(diffSummary: string): DiffSummary {
+  try {
+    return JSON.parse(diffSummary) || {};
+  } catch (err) {
+    throw err;
+  }
+}
+
+function getPreviousSequence(versionHistory: Version[], version: Version) {
+  let previousSequence = 0;
+  for (const v of versionHistory) {
+    if (v.status === "pending_download") {
+      continue;
+    }
+    if (v.parentSequence < version.parentSequence) {
+      previousSequence = v.parentSequence;
+      break;
+    }
+  }
+  return previousSequence;
+}
+
+function ViewDiffButton(props: Props) {
+  const { selectedApp } = useSelectedApp();
+  const { data: isHelmManagedResponse } = useIsHelmManaged();
+  const isHelmManaged = isHelmManagedResponse?.isHelmManaged || false;
+
+  // TODO: flatten in selector
+  const showViewDiffButton =
+    !isHelmManaged && !selectedApp?.downstream.gitops?.isConnected;
+  const showDiffSummaryError =
+    props.version?.diffSummaryError?.length > 0 ? true : false;
+  const numberOfFilesChanged = props.version?.diffSummary
+    ? unmarshallDiffSummary(props.version.diffSummary)?.filesChanged || 0
+    : 0;
+
+  return (
+    <>
+      {showDiffSummaryError && (
+        <div className="flex flex1 alignItems--center">
+          <span className="u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-textColor--bodyCopy">
+            Unable to generate diff{" "}
+            <span
+              className="link"
+              onClick={() =>
+                props.onWhyUnableToGeneratedDiffClicked(props.version)
+              }
+            >
+              Why?
+            </span>
+          </span>
+        </div>
+      )}
+
+      {numberOfFilesChanged > 0 && (
+        <div className="u-fontSize--small u-fontWeight--medium u-lineHeight--normal">
+          <div className="DiffSummary u-marginRight--10">
+            <span className="files">{numberOfFilesChanged} files changed </span>
+            {showViewDiffButton && (
+              <span
```

Comment:
can probably tighten this up better but i didn't want to make too many changes right now. I will revisit when i start making the hooks